pytorch_lightning/core/saving.py

```diff
@@ -372,10 +373,20 @@ def save_hparams_to_yaml(config_yaml, hparams: Union[dict, Namespace]) -> None:
                     OmegaConf.save(OmegaConf.create(hparams), fp, resolve=True)
                 return
 
-    # saving the standard way
     assert isinstance(hparams, dict)
+    hparams_allowed = {}
+    # drop paramaters which contain some strange datatypes as fsspec
+    for k, v in hparams.items():
+        try:
+            yaml.dump(v)
+        except TypeError as err:
+            warn(f"Skipping '{k}' parameter because it is not possible to safely dump to YAML.")
```

Comment:
Not sure if we could still save the class name here to help the user understand the type?

But we would then have to filter it on loading

Reply:
but what if it is not a class, can be an enum with some hidden things inside...
so just put there string `hparams[k] = str(v)`?

Reply:
either that or really just `hparams[k] = type(v).__name__`